src/osd/OSD.cc

```diff
@@ -3264,8 +3266,10 @@ void OSD::build_past_intervals_parallel()
 
       epoch_t start, end;
       if (!pg->_calc_past_interval_range(&start, &end, superblock.oldest_map)) {
-        if (pg->info.history.same_interval_since == 0)
+        if (pg->info.history.same_interval_since == 0) {
           pg->info.history.same_interval_since = end;
+          fixed.insert(pg);
```

Comment:
maybe we can have
```c++
pis.start = pis.end = 0;
pis.same_interval_since = end;
```
and skip the interval calculation later on. and do not set `pg->dirty_big_info` when preparing the transactions if the `pis.begin == pis.end`.

@dzafman what do you think? shall we write the pg info if the pg has no past intervals?